Add support for eslintrc.json seti-ui icon

adaex opened the pull request:
This PR follows the steps from [CONTRIBUTING.md](https://github.com/microsoft/vscode/blob/main/extensions/theme-seti/CONTRIBUTING.md) to pull in updates from seti-ui, adding an already existing icon for `eslintrc.js` file to `eslintrc.json`

[jesseweed/seti-ui](https://github.com/jesseweed/seti-ui) already supports the icon for `eslintrc.json` file. [View line 554.](https://github.com/jesseweed/seti-ui/blob/2b078f89a535a9f6dda47817bac01f02a4350078/styles/components/icons/mapping.less#L554)

Screenshot of `preview.html`:

![image](https://user-images.githubusercontent.com/29560987/117577110-091c1200-b11b-11eb-934d-1d80e4eebd4d.png)

aeschli commented:
Can you explain how you got to these changes?

I just updated to the latest seti-theme: https://github.com/microsoft/vscode/commit/86bc05155f8c3b1b2682556c28cf0e6f96bfe220

- have the main branches of `https://github.com/jesseweed/seti-ui` and `https://github.com/microsoft/vscode` cloned in the same parent folder
- in the `seti-ui` folder, run `npm install` and `npm run prepublishOnly`. This will generate updated icons and fonts.
- in the `vscode/extensions/theme-seti` folder run  `npm run update`. This will launch the [icon theme update script](build/update-icon-theme.js) that updates the theme as well as the font based on content in `seti-ui`.
- to test the icon theme, look at the icon preview as described above.

I updated [CONTRIBUTING.md](https://github.com/microsoft/vscode/blob/main/extensions/theme-seti/CONTRIBUTING.md) with these clarifications

adaex commented:
@aeschli , In fact, this problem puzzled me at first, but now I seem to find the reason why the `eslintrc.json` file icon is not generated.

Let me repeat that.

1. In line 52 of the file `extensions/json/package.json`, we can see that the `extensions` of `jsonc` is assigned `.jsonc`, `.eslintrc`, `.eslintrc.json`, etc.

![image](https://user-images.githubusercontent.com/29560987/117954941-1703aa00-b34a-11eb-8d24-26974b6438c8.png)

2. In `extensions/theme-seti/build/update-icon-theme.js`, we can see that `preferredDef` attempts to assign values through `extensions` and `filenames`.

![image](https://user-images.githubusercontent.com/29560987/117957152-58955480-b34c-11eb-9c56-c80f8271e5c0.png)

> For `jsonc`, the extensions `.jsonc` does not exist in `seti-ui`, so the value of `preferredDef` is assigned to the icon of `.eslintrc`.

3. According to the following code, we can find that the file allocation of `.eslintrc` and `.eslintrc.json` has been deleted.

![image](https://user-images.githubusercontent.com/29560987/117957932-1b7d9200-b34d-11eb-949b-086868283414.png)

If we still don't understand, we can check the logic. Follow the screenshot below to add some code.

![image](https://user-images.githubusercontent.com/29560987/117960842-05bd9c00-b350-11eb-8022-6254ced70d54.png)

Run `npm run update`, Here is the output:

```
node ➜ /workspaces/vscode/extensions/theme-seti (eslint-icon ✗) $ npm run update

> vscode-theme-seti@1.0.0 update /workspaces/vscode/extensions/theme-seti
> node ./build/update-icon-theme.js

Reading from ../../../seti-ui/styles/_fonts/seti.less
lang: jsonc  preferredDef: _eslint
remove the extension association: eslintrc
remove the extension association: eslintrc.json
written ./icons/vs-seti-icon-theme.json
updated ./cgmanifest.json
Updated to jesseweed/seti-ui@2b078f8 (2021-05-05)
```

We can verify that `jsonc` is assigned as the `_eslint` icon, and finally remove the file icon assignment of `eslintrc` and `eslintrc.json`.

My current solution is to ignore this part of the operation if the language has been defined in `inheritIconFromLanguage`.

![image](https://user-images.githubusercontent.com/29560987/117968778-26d6ba80-b359-11eb-8e45-454652a461f5.png)

aeschli commented:
Thanks @adaex !
